### PR TITLE
S722982: Complete unit test coverage for current functionality of EmitApiPython

### PR DESCRIPTION
**Pull request by cchan-ansys**

### CODE REVIEW

**Comment by Samuelopez-ansys** [comment]
Hi @cchan-ansys ,

I think you are trying to merge an old branch that was already merged in the main. You probably made some modifications to this branch, as a best practice you should merge the main into your local branch before doing a Pull request. 

I could show you how using PyCharm almost everything is automated.

File: pyaedt/emit.py [hunk 1]
```diff
@@ -1,16 +1,439 @@
...
+
+        Examples
+        ----------
+        >>> tx_frequencies = domain.get_interferer_channel_frequencies()
+        """
+
+        return list(self._obj.interferer_channel_frequencies)
+
+    @pyaedt_function_handler()
+    def get_instance_count(self):
```
Comment (MaxJPRey) [inline pyaedt/emit.py:175]: @cchan-ansys This could be done using `property`.

File: pyaedt/emit.py [hunk 2]
```diff
@@ -1,16 +1,439 @@
...
+        -------
+
+        Examples
+        ----------
+        >>> tx_names = domain.get_interferer_names()
+        """
+        return list(self._obj.interferer_names)
+
+    @pyaedt_function_handler()
+    def get_interferer_band_names(self):
```
Comment (MaxJPRey) [inline pyaedt/emit.py:148]: @cchan-ansys This could be done using `property`.

File: pyaedt/emit.py [hunk 3]
```diff
@@ -1,16 +1,439 @@
...
+        -------
+
+        Examples
+        ----------
+        >>> tx_band_names = domain.get_interferer_band_names()
+        """
+        return list(self._obj.interferer_band_names)
+
+    @pyaedt_function_handler()
+    def get_interferer_channel_frequencies(self):
```
Comment (MaxJPRey) [inline pyaedt/emit.py:161]: @cchan-ansys This could be done using `property`.

File: pyaedt/emit.py [hunk 4]
```diff
@@ -1,16 +1,439 @@
...
+        -------
+
+        Examples
+        ----------
+        >>> rx_freq = domain.receiver_channel()
+        """
+        return self._obj.receiver_channel
+
+    @pyaedt_function_handler()
+    def get_interferer_names(self):
```
Comment (MaxJPRey) [inline pyaedt/emit.py:135]: @cchan-ansys This could be done using a `property`.

**Review by MaxJPRey — DISMISSED** [review]
Some methods could be modified into properties.

File: pyaedt/emit.py [hunk 5]
```diff
@@ -12,6 +12,198 @@
...
+    def __init__(self):
+        self.emit_api = mod.EmitApi()
+        self._obj = mod.InteractionDomain()
+        print(str(type(self._obj)))
+        try:
+            self._obj = mod.InteractionDomain()
+        except NameError:
+            raise ValueError("An Emit object must be initialized before an Interaction_Domain object is generated.")
+
+    @pyaedt_function_handler()
```
Comment (Samuelopez-ansys) [inline pyaedt/emit.py:38]:
```suggestion
```

**Review by Samuelopez-ansys — APPROVED** [review]
LGTM

**Review by Samuelopez-ansys — APPROVED** [review]
LGTM

**Pull request merged**